src/main/java/com/bootcamp/demo/restapi/BookingsController.java

```diff
@@ -0,0 +1,135 @@
+        LinkedHashSet<Booking> bookingsByUserId = new LinkedHashSet<>();
+        try {
+            bookingsByUserId = bookingService.getBookings(userId);
+            LOGGER.info("GET BOOKINGS BY userID - API endpoint invoked");
+            return new ResponseEntity<>(bookingsByUserId, HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            e.printStackTrace();
+            return new ResponseEntity<>(bookingsByUserId, HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @GetMapping("/getBooking/{id}")
+    @ResponseBody
+    public ResponseEntity<Object> getBooking(@PathVariable(value = "id") final UUID id) {
+        Booking booking = null;
+        try {
+            booking = bookingService.getBookingByID(id);
+            LOGGER.info("GET BOOKING BY ID - API endpoint invoked");
+            return new ResponseEntity<>(booking, HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            e.printStackTrace();
+            return new ResponseEntity<>(booking, HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @GetMapping("/showAllBookings")
+    @ResponseBody
+    public ResponseEntity<LinkedHashSet<Booking>> getAllBookings() {
+        LinkedHashSet<Booking> bookings = new LinkedHashSet<>();
+        try {
+            bookings = bookingService.getAllBookings();
+            return new ResponseEntity<>(bookings, HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException e) {
+            e.printStackTrace();
+            return new ResponseEntity<>(bookings, HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @PutMapping("/delete")
+    public ResponseEntity<Object> deleteBooking(@RequestParam final UUID id) {
+        try {
+            bookingService.deleteBooking(id);
+            LOGGER.info("DELETE BOOKING - API endpoint invoked");
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            e.printStackTrace();
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    @PutMapping("/update")
+    public ResponseEntity<Object> updateBooking(@RequestParam final UUID id, @RequestParam("start")
+    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)  LocalDateTime endDate, PaymentStatus status) {
+        try {
+            bookingService.updateBooking(id, endDate, status);
+            LOGGER.info("UPDATE BOOKING - API endpoint invoked");
+            return new ResponseEntity<>(HttpStatus.OK);
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            e.printStackTrace();
+            return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    public double endOfBooking(Booking booking, Scooter scooter) throws ExecutionException, InterruptedException {
```

Comment:
Will this be called as part of `updateBooking` or of another method? If in `updateBooking`, perhaps a more intuitive name should be used (and not one that is so general).
Also, you should renamed this method to better highlight what it does.